sinks/graphite.py

```diff
@@ -163,21 +168,18 @@ def _write_metric(self, metric):
 
 
 def main():
-
     # Intialize from our arguments
     graphite = GraphiteStore(*sys.argv[1:])
 
+    METRICS_PER_FLUSH = 5000
+
     # Get all the inputs
-    while True:
-        try:
-            graphite.append(raw_input().strip())
-        except EOFError:
-            break
-
-    # Flush
-    graphite.logger.info("Outputting %d metrics", len(graphite.metrics))
-    graphite.flush()
-    graphite.close()
+    for line in sys.stdin:
```

Comment:
There seem to be minor differences between raw_input() and sys.stdin, but both work in this case.